src/Compilers/Core/Portable/Syntax/SyntaxTreeExtensions.cs

```diff
@@ -94,5 +95,19 @@ private static int FindFirstDifference(string s1, string s2)
             }
             return (n1 == n2) ? -1 : n + 1;
         }
+
+        /// <summary>
+        /// Returns <c>true</c> if the provided position is in a hidden region inaccessible to the user.
+        /// </summary>
+        public static bool IsHiddenPosition(this SyntaxTree tree, int position, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            if (!tree.HasHiddenRegions())
+            {
+                return false;
+            }
+
+            var lineVisibility = tree.GetLineVisibility(position, cancellationToken);
```

Comment:
Yes, line directive map realizes the source text, but only if there are any line directives in tree, which should be rare.